client/components/select-vertical/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { useTranslate } from 'i18n-calypso';
+import { useCallback, useState } from 'react';
+import { UseQueryOptions } from 'react-query';
+import { useDebounce } from 'use-debounce';
+import FormLabel from 'calypso/components/forms/form-label';
+import { useSiteVerticalsQuery } from 'calypso/data/site-verticals';
+import SuggestionSearch from './suggestion-search';
+import type { Vertical } from './types';
+import type { SiteVerticalsResponse, SiteVerticalsQueryParams } from 'calypso/data/site-verticals';
+
+interface Props {
+	isSkipSynonyms?: boolean;
+}
+
+const SelectVertical: React.FC< Props > = ( { isSkipSynonyms } ) => {
+	const translate = useTranslate();
+	const [ searchTerm, setSearchTerm ] = useState( '' );
+	const [ debouncedSearchTerm ] = useDebounce( searchTerm, 300 );
+	const { data: suggestions } = useSiteVerticalsQuery(
+		{ term: debouncedSearchTerm, skip_synonyms: isSkipSynonyms } as SiteVerticalsQueryParams,
```

Comment:
I believe these `as SiteVerticalsQueryParams` and `as UseQueryOptions` are not necessary here, as they already conform to the respective types.

Reply:
Type-checking making me overcautious.